src/basic/parse-util.c

```diff
@@ -402,13 +430,17 @@ int safe_atou_full(const char *s, unsigned base, unsigned *ret_u) {
 }
 
 int safe_atoi(const char *s, int *ret_i) {
+        unsigned base = 0;
         char *x = NULL;
         long l;
 
         assert(s);
 
+        s += strspn(s, WHITESPACE);
```

Comment:
Is this a good idea? Why should we allow leading whitespace here?

Reply:
well, not sure if its, but strtol() does it implicitly when parsing normally, and that's documented. I wanted to make this as little invasive as possible, so I tried to add the 0b + 0o support in a fashion that behaves as similar as possible to how 0x is already handled.

(I must say, given all the weird behaviour of strtol() that keeps revealing itself I am tempted to just replace it altogether by our own code that parses only what we want it to parse, and doesn't eat whitspace and so on. it feels right now we need more code to block all the stuff it does we don't want than it would take to write our own implementation that only does what we want. but maybe let's leave that for a later PR)

Reply:
> replace it altogether by our own code that parses only what we want it to parse

Indeed.

> let's leave that for a later PR

Ack.